join.py

```diff
@@ -0,0 +1,20 @@
+import itertools
+
+
+def join(*lists, sep='-'):
+    """
+    The function receives an unlimited number of lists and another parameter called sep (optional)
+    and returns one list consisting of all the lists received as parameters. If the sep parameter is provided,
+    it threads it as an object between any two lists, and if it is not provided, it threads the character "-" instead.
+    :param lists: The lists to be merged into one list.
+    :param sep: The separator between the lists.
+    :return: The merged list.
+    """
+    if not lists:
+        return None
```

Comment:
This behavior should be documented, I would return an empty list and not None  - will be easier to handle when other functions use it.
Read this - 
https://betterprogramming.pub/please-stop-returning-none-from-python-functions-ff98e492b4b0
And think about what the outputs of these inputs will be - 
[]
[[],[],[],]

Reply:
Hi,
The requirements of the exercise are as follows:
הקריאה join() תחזיר None או שגיאה, לבחירתכם
So I returned None.